pkg/crypto/crypto_test.go

```diff
@@ -84,27 +84,90 @@ func decryptConfigMap(configMap string, keyPath string, noncePath string) string
 	//The keys are base64 encoded. Decode it before passing to encryption function
 	keyDecoded, err := b64.StdEncoding.DecodeString(string(key))
 	if err != nil {
-		panic(err.Error())
+		return "", err
 	}
 	nonceDecoded, err := b64.StdEncoding.DecodeString(string(nonce))
 	if err != nil {
-		panic(err.Error())
+		return "", err
 	}
 
 	block, err := aes.NewCipher(keyDecoded)
 	if err != nil {
-		panic(err.Error())
+		return "", err
 	}
 
 	aesgcm, err := cipher.NewGCM(block)
 	if err != nil {
-		panic(err.Error())
+		return "", err
 	}
 
 	plaintextBytes, err := aesgcm.Open(nil, nonceDecoded, sDec, nil)
 	if err != nil {
-		panic(err.Error())
+		return "", err
 	}
 
-	return string(plaintextBytes)
+	return string(plaintextBytes), nil
+}
+
+func TestDecryptConfigMapWrongkey(t *testing.T) {
+
+	testString := "Sample String to Test Symmetric Encryption"
+
+	dir1, err := ioutil.TempDir("", "TestEncryption-2")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.RemoveAll(dir1)
+
+	keyPath := dir1 + "symmKeyFile"
+	wrongkeyPath := dir1 + "wrongsymmKeyFile"
+
+	key := make([]byte, 32)
+	if _, err := io.ReadFull(rand.Reader, key); err != nil {
+		t.Error(err)
+	}
+	wrongkey := make([]byte, 32)
+	if _, err := io.ReadFull(rand.Reader, wrongkey); err != nil {
+		t.Error(err)
+	}
+
+	//Encode the Key
+	keyEnc := b64.StdEncoding.EncodeToString(key)
+	//encode wrong key
+	wrongkeyEnc := b64.StdEncoding.EncodeToString(wrongkey)
+
+	err = ioutil.WriteFile(keyPath, []byte(keyEnc), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+	err = ioutil.WriteFile(wrongkeyPath, []byte(wrongkeyEnc), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+
+	noncePath := dir1 + "nonceFile"
+	nonce := make([]byte, 12)
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		t.Error(err)
+	}
+
+	//Encode the nonce
+	nonceEnc := b64.StdEncoding.EncodeToString(nonce)
+
+	err = ioutil.WriteFile(noncePath, []byte(nonceEnc), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+
+	encConfigMap, err := EncryptConfigMap([]byte(testString), keyPath, noncePath)
+
+	if err != nil {
+		t.Errorf("Failed to encrypt configMap %v", err)
+	}
+
+	wrongdecryptedConfigMap ,err := decryptConfigMap(encConfigMap, wrongkeyPath, noncePath)
```

Comment:
everything else looks okay, if you can compare the error you are expecting will be great after this function call makes the test case complete.

Reply:
> 
> 
> everything else looks okay, if you can compare the error you are expecting will be great after this function call makes the test case complete.

@mkumatag : Added this change. Now err return value is getting checked. Please review and let me know if you have further comments.